File: capturar-front/src/app/features/panel/data-access/photo-upload.models.ts
export interface PresignedUrlsRequest {
  eventId: number;
  fileNames: string[];
}

export interface PresignedUrlItem {
  fileName: string;
  objectKey: string;
  uploadUrl: string;
}

export interface PresignedUrlsResponse {
  eventId: number;
  files: PresignedUrlItem[];
}

export interface ConfirmUploadFileItem {
  fileName: string;
  objectKey: string;
  sizeBytes: number;
}

export interface ConfirmUploadRequest {
  eventId: number;
  files: ConfirmUploadFileItem[];
}

export interface ConfirmUploadResponse {
  eventId: number;
  savedCount: number;
  photoIds: number[];
  missingFiles: ConfirmUploadFileItem[];
}

export interface ProxyUploadResponse {
  eventId: number;
  objectKey: string;
  fileName: string;
  sizeBytes: number;
}

export interface DownloadPhotoResponse {
  photoId: number;
  downloadUrl: string;
  expiresInSeconds: number;
}

export interface PhotoProcessingItem {
  photoId: number;
  fileName: string;
  isProcessed: boolean;
  isFailed: boolean;
  failureReason?: string | null;
  createdAt: string;
}

export interface PhotoProcessingStatusResponse {
  eventId: number;
  totalPhotos: number;
  processedPhotos: number;
  failedPhotos: number;
  pendingPhotos: number;
  progressPercent: number;
  recentPhotos: PhotoProcessingItem[];
}

export interface BatchProcessingStatusRequest {
  photoIds: number[];
}

export interface PhotographerGalleryPhotoItem {
  photoId: number;
  fileName: string;
  isProcessed: boolean;
  isFailed: boolean;
  failureReason?: string | null;
  thumbnailUrl: string;
  previewUrl: string;
  tags: string[];
  createdAt: string;
}

export interface UpdatePhotoTagsResponse {
  eventId: number;
  photoId: number;
  tags: string[];
}

export interface PhotographerGalleryResponse {
  eventId: number;
  page: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
  items: PhotographerGalleryPhotoItem[];
}

File: admin-base-api/Admin.WebApi/Models/PhotoUploadModels.cs
using Microsoft.AspNetCore.Http;

namespace Admin.WebApi.Models;

public class PresignedUrlsRequest
{
    public int EventId { get; set; }
    public List<string> FileNames { get; set; } = new();
}

public class PresignedUrlItem
{
    public string FileName { get; set; } = string.Empty;
    public string ObjectKey { get; set; } = string.Empty;
    public string UploadUrl { get; set; } = string.Empty;
}

public class PresignedUrlsResponse
{
    public int EventId { get; set; }
    public List<PresignedUrlItem> Files { get; set; } = new();
}

public class ConfirmUploadRequest
{
    public int EventId { get; set; }
    public List<UploadedFileItem> Files { get; set; } = new();
}

public class UploadedFileItem
{
    public string FileName { get; set; } = string.Empty;
    public string ObjectKey { get; set; } = string.Empty;
    public long SizeBytes { get; set; }
}

public class ConfirmUploadResponse
{
    public int EventId { get; set; }
    public int SavedCount { get; set; }
    public List<int> PhotoIds { get; set; } = new();
    public List<UploadedFileItem> MissingFiles { get; set; } = new();
}

public class DownloadPhotoResponse
{
    public int PhotoId { get; set; }
    public string DownloadUrl { get; set; } = string.Empty;
    public int ExpiresInSeconds { get; set; }
}

public class PhotoProcessingStatusResponse
{
    public int EventId { get; set; }
    public int TotalPhotos { get; set; }
    public int ProcessedPhotos { get; set; }
    public int FailedPhotos { get; set; }
    public int PendingPhotos { get; set; }
    public int ProgressPercent { get; set; }
    public List<PhotoProcessingItemDto> RecentPhotos { get; set; } = new();
}

public class BatchProcessingStatusRequest
{
    public List<int> PhotoIds { get; set; } = new();
}

public class PhotoProcessingItemDto
{
    public int PhotoId { get; set; }
    public string FileName { get; set; } = string.Empty;
    public bool IsProcessed { get; set; }
    public bool IsFailed { get; set; }
    public string? FailureReason { get; set; }
    public DateTime CreatedAt { get; set; }
}

public class ProxyUploadRequest
{
    public int EventId { get; set; }
    public string ObjectKey { get; set; } = string.Empty;
    public string FileName { get; set; } = string.Empty;
    public IFormFile? File { get; set; }
}

public class ProxyUploadResponse
{
    public int EventId { get; set; }
    public string ObjectKey { get; set; } = string.Empty;
    public string FileName { get; set; } = string.Empty;
    public long SizeBytes { get; set; }
}

public class PhotographerGalleryPhotoDto
{
    public int PhotoId { get; set; }
    public string FileName { get; set; } = string.Empty;
    public bool IsProcessed { get; set; }
    public bool IsFailed { get; set; }
    public string? FailureReason { get; set; }
    public string ThumbnailUrl { get; set; } = string.Empty;
    public string PreviewUrl { get; set; } = string.Empty;
    public List<string> Tags { get; set; } = new();
    public DateTime CreatedAt { get; set; }
}

public class UpdatePhotoTagsRequest
{
    public List<string> Tags { get; set; } = new();
}

public class PhotographerGalleryResponse
{
    public int EventId { get; set; }
    public int Page { get; set; }
    public int PageSize { get; set; }
    public int TotalCount { get; set; }
    public int TotalPages { get; set; }
    public List<PhotographerGalleryPhotoDto> Items { get; set; } = new();
}
